Projects/Project10 Names to Faces/Project10 Names to Faces/ViewController.swift

```diff
@@ -91,27 +167,38 @@ class ViewController: UICollectionViewController, UIImagePickerControllerDelegat
         present(ac, animated: true)
     }
     
-    func renamePerson(at indexPath: IndexPath) {
-        let ac = UIAlertController(title: "Rename Person", message: nil, preferredStyle: .alert)
-        ac.addTextField()
-        ac.addAction(UIAlertAction(title: "Submit", style: .default) {
-            [weak self, weak ac] _ in
-            guard let newName = ac?.textFields?[0].text else { return }
-            self?.people[indexPath.item].name = newName
-            self?.collectionView.reloadData()
-        })
-        ac.addAction(UIAlertAction(title: "Cancel", style: .cancel))
-        present(ac, animated: true)
+}
+
+
+// MARK: - EXTENSIONS
+
+extension ViewController: UIImagePickerControllerDelegate, UINavigationControllerDelegate {
+    @objc func addNewPerson() {
+        let picker = UIImagePickerController()
+        picker.allowsEditing = true
+        picker.delegate = self
+        present(picker, animated: true)
     }
     
-    func deletePerson(at indexPath: IndexPath) {
-        let ac = UIAlertController(title: "Are you sure?", message: "Once deleted it will be gone forever.", preferredStyle: .alert)
-        ac.addAction(UIAlertAction(title: "Yes", style: .default) {
-            [weak self] _ in
-            self?.people.remove(at: indexPath.item)
-            self?.collectionView.reloadData()
-        })
-        ac.addAction(UIAlertAction(title: "Cancel", style: .cancel))
-        present(ac, animated: true)
+    @objc func addNewPersonByCamera() {
+        let picker = UIImagePickerController()
+        picker.sourceType = .camera
+        picker.allowsEditing = true
+        picker.delegate = self
+        present(picker, animated: true)
+    }
+    
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [UIImagePickerController.InfoKey : Any]) {
+        guard let image = info[.editedImage] as? UIImage else { return }
+        let imageName = UUID().uuidString
+        let imagePath = getDocumentsDirectory().appendingPathComponent(imageName)
+        if let jpegData = image.jpegData(compressionQuality: 0.8)  {
+            try? jpegData.write(to: imagePath)
+        }
```

Comment:
Might want to do this in a try catch ... in a real app you would want to make sure you had the ability to inform on a fail like this.